src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -945,28 +947,47 @@ class OpenShiftDSL implements Serializable {
         OpenShiftResourceSelector svcSel = this.selector("svc", svcName);
         String ip = (String)svcSel.object().get("spec").get("clusterIP");
         String port = (String)svcSel.object().get("spec").get("ports").get(0).get("port");
+        String ns = (String)currentContext.getProject();
+        String svcAddress = svcName+"."+ns+".svc.cluster.local";
         int i = 0;
         while (i < 5) {
             i++;
-            InetSocketAddress address = new InetSocketAddress(ip, Integer.parseInt(port));
-            Socket socket = null;
-            try {
-                socket = new Socket();
-                socket.connect(address, 2500);
-                logToTaskListener("Connected to service " + svcName + " via IP " + ip + " and port " + port);
-                return true;
-            } catch (IOException e) {
-                logToTaskListener("IOException " + e.getMessage() + " connecting to service " + svcName);
-                LOGGER.log(Level.FINE, "verifyService", e);
-                try {
-                    Thread.sleep(2500);
-                } catch (InterruptedException e1) {
+            if (ip == "None"){ 
+                InetAddress inetAddress = null;
+                try{
+                    inetAddress = InetAddress.getByName(svcName);
```

Comment:
@waveywaves I've got a question in to our networking team on whether the dns lookup should work even if the jenkins pod is in a separate namespace than the headless service

Reply:
I think it should be want to make sure.

In the interim, if you could construct such a test case and confirm/deny that fact, do that and respond here.

Reply:
I will check if it works with services in other namespaces as well once and update it here along with the README.